tests: spread system user autoimport

fgimenez opened the pull request:
This PR uses a model and system-user assertion signed by the same authority-id in order to make the system-user assertion importable. I still don't have available a system-user assertion signed by the canonical authority-id (already contacted with @emgee) so right now the PR replaces the canonical-signed model used to create the image, we probably want to keep it (and use a system-user assertion signed by canonical) or modify it adding a `system-user-authority` field.

zyga commented on tests/lib/prepare.sh:
```diff
@@ -233,6 +233,51 @@ linux /vmlinuz root=$ROOT ro init=$IMAGE_HOME/reflash.sh console=ttyS0
+        # deactivate swap
+        swapoff -a
+        # create new primary partition occupying all the /dev/sdb disk
+        echo -e "o\nn\np\n1\n\n\nw" | fdisk /dev/sdb
```
I'm amazed that after several years since I used it first this is still the best way to create partitions in scripts. May forth backwards school live on :-)  (nothing against your work Federico, just an observation)

fgimenez replied:
@zyga yeap, doesn't feel right, any alternatives are more than welcome :)

zyga commented on tests/lib/state.sh:
```diff
@@ -0,0 +1,15 @@
+remove_users_from_state(){
+    systemctl stop snapd.service
+    cp /var/lib/snapd/state.json ./state.json.back
+    sed -i 's/"users":\[[^]]*\]/"users":null/' /var/lib/snapd/state.json
```
This feels fragile, I'd rather use python to delete users from the state.

niemeyer replied:
+1 to jq.

niemeyer commented on tests/lib/state.sh:
```diff
@@ -0,0 +1,15 @@
+#!/bin/sh
+remove_users_from_state(){
+    systemctl stop snapd.service
+    cp /var/lib/snapd/state.json ./state.json.back
```
Can we please separate this into an explicit action called `save_state` used on prepare, name the file as `state.json.saved`, and have that as an independent pair of `restore_state`? Then it's more explicit and clear when the state is saved and when it's restored.  The removal function can then do just that: remove users from state.

fgimenez replied:
nice, thanks! done

fgimenez commented:
Thanks for your reviews, I've updated the PR with the suggested changes and also replaced fdisk with sfdisk for the block device partitioning (thanks to @chipaca! :))

fgimenez commented:
Please do not merge yet regardless of the results, the test is being flaky and I'm still debugging it.

niemeyer commented:
Tests (seemingly) broken.

fgimenez commented:
This is still failing eventually, the actual error is:
```
Nov 22 16:03:07 localhost.localdomain systemd-udevd[1057]: Process '/usr/bin/unshare -m /usr/bin/snap auto-import --mount=/dev/sdb1' failed with exit code 1.
```

This is the complete `journalctl -xe` output after the error https://trello-attachments.s3.amazonaws.com/56d05584c5822af1cb24e16f/58341e78a9b3fbeef88de35c/aa87079deec0a2f764c37e29140ff21f/journalctl.out.error

zyga commented on run-checks:
```diff
@@ -135,7 +135,7 @@ if [ "$SPREAD" = 1 ]; then
     ( cd $TMP_SPREAD && curl -s -O https://niemeyer.s3.amazonaws.com/spread-amd64.tar.gz && tar xzvf spread-amd64.tar.gz )
 
-    spread -v linode:
+    spread -vv linode:ubuntu-core-16-64:tests/main/ubuntu-core-system-user-autoimport
```
Hmm, are you sure we want this change?

fgimenez replied:
> Hmm, are you sure we want this change?

@zyga no way! :) but I think this is an outdated diff, it's already removed from the changeset, could you pls confirm? Thanks! :)

fgimenez commented:
This is still failing randomly (2/10 failures in last execution), closing for now.
